lib/src/Uncertainty/Algorithm/Analytical/Analytical.cxx

```diff
@@ -131,6 +146,18 @@ void Analytical::run()
   Point value(standardEvent.getImplementation()->getFunction().operator()(origin));
 
   result_.setIsStandardPointOriginInFailureSpace(event_.getOperator().compare(value[0], event_.getThreshold()));
+  
+  /* check if result is valid */
+  Point physicalSpaceDesignPoint(event_.getImplementation()->getAntecedent().getDistribution().getInverseIsoProbabilisticTransformation().operator()(standardSpaceDesignPoint));
+  
+  Point valuePhysicalSpaceDesignPoint(event_.getImplementation()->getFunction().operator()(physicalSpaceDesignPoint));
+  
+  const Scalar residual = std::abs(valuePhysicalSpaceDesignPoint[0] - event_.getThreshold()); 
+  
+  if (!(residual <= limitStateTolerance_))
```

Comment:
- It seems to me that, if the value of the threshold has a very large magnitude or is very close to zero, it might be difficult to chose a relevant value of the tolerance. For example, if the threshold is a pressure, the order of magnitude might be close to $10^9$. In this case, an absolute tolerance equal to 0.01 is _extremely_ stringent.
- We cannot use a relative tolerance neither, because the user-defined threshold might be zero.
- One way to solve the problem is to use a mixed absolute-relative criteria. Introducing a new `limitStateRelativeTolerance_` parameter, this goes like:

```suggestion
  if (!(std::abs(valuePhysicalSpaceDesignPoint[0] - event_.getThreshold()) <= limitStateAbsoluteTolerance_ + limitStateRelativeTolerance_ * std::abs(event_.getThreshold())))
```

Reply:
I agree with you about the fact that the threshold is problem-dependent. IMO, in most of the real world problems, the tolerance is given by expert so it has to be tuned before using the algorithm. I'm not convinced that mixing the absolute and relative tolerance in a composite criterion will be the most easy to understand for the user. Maybe we can discuss in depth of this during the next technical committee?